### PR TITLE
✨(cli) allow to force push for database backends

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

When bulk importing documents in ES, if a document is badly formatted (e.g. with mistyped fields), the bulk importation stops after the first chunk importation failure.

## Proposal

We now allow to force the importation regardless of indexation errors.

### CODE REVIEW

File: src/ralph/backends/database/es.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def to_documents(self, stream, get_id):
                 "_source": item,
             }
 
-    def put(self, chunk_size=500):
+    def put(self, chunk_size=500, force=False):
```
Comment (lunika) [inline src/ralph/backends/database/es.py:50]: `silent_errors` or `ignore_errors` should be more relevant IMO. I understand `force` like a `git push -f` and it's not the same meaning for me.

File: tests/backends/database/test_es.py [hunk 2]
```diff
@@ -126,3 +128,62 @@ def test_write_method(es, fs, monkeypatch):
...
+    )
+
+    assert len(es.search(index=ES_TEST_INDEX)["hits"]["hits"]) == 0
+
+    database = ESDatabase(
+        hosts=ES_TEST_HOSTS,
+        index=ES_TEST_INDEX,
+    )
+
+    # By default, we sould raise an error and stop the importation
```
Comment (SergioSim) [inline tests/backends/database/test_es.py:153]:
```suggestion
    # By default, we should raise an error and stop the importation
```

**Review by SergioSim — COMMENTED** [review]
This looks great!

File: tests/backends/database/test_es.py [hunk 3]
```diff
@@ -126,3 +128,62 @@ def test_write_method(es, fs, monkeypatch):
...
+    """Test ES put method with badly formatted data when the force mode is active"""
+    # pylint: disable=invalid-name,unused-argument
+
+    records = [{"id": idx, "count": random.randint(0, 100)} for idx in range(10)]
+    # Patch a record with a non-expected type for the count field (should be
+    # assigned as long)
+    records[2].update({"count": "wrong"})
+
+    monkeypatch.setattr(
+        "sys.stdin", StringIO("\n".join([json.dumps(record) for record in records]))
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply tests/backends/database/test_es.py:172]: As JSONs are generated using the `json.dumps` method, I think we can consider generated data as valid no?

[the remark this answers is not in the document]
Comment (SergioSim) [reply tests/backends/database/test_es.py:172]: Good point, yes indeed, if we always read what `json.dumps` outputs then it's fine

File: src/ralph/backends/database/es.py (hunk 1 above)
Comment (jmaupetit) [reply src/ralph/backends/database/es.py:50]: When forcing a `git push`, it means push this regardless of potential errors occurring without the `--force` flag. So it's basically the same to me.

Comment (lunika) [reply src/ralph/backends/database/es.py:50]: No with `git push --force` you bypass a security, you force to overwrite a remote ref that is not an ancestor of the local ref. It has nothing to do with ignoring errors.

**Review by quitterie-lcs — APPROVED** [review]
Modifications have been reviewed, ready for merging!

File: src/ralph/backends/database/es.py (hunk 1 above)
Comment (jmaupetit) [reply src/ralph/backends/database/es.py]: After IRL conversation we decided to introduce a new `ignore-errors` option to the `push` command.

**Comment by jmaupetit** [comment]
@lunika I've addressed your comment. Feel free to give this PR a new try. Thanks! :pray:

**Review by lunika — APPROVED** [review]

**Pull request merged**